### PR TITLE
Add in-home chime support to CLI

### PR DESCRIPTION
**Pull request by briangoldstein**
Added a command `in-home-chime` which supports viewing the current in-home chime settings for a device as well as changing any of the available settings using multiple arguments.

Sample output of `ring-doorbell in-home-chime --device-name "Front Door"`

> Name:       Front Door
> ID:         987652
> Type:       Digital
> Enabled:    True
> Duration:   5

Output of `ring-doorbell in-home-chime --help`

> Usage: ring-doorbell in-home-chime [OPTIONS]
> 
>   View and manage the Doorbell in-home chime. To see the current in-home chime
>   status of a device, only pass the device name.
> 
> Options:
>   -dn, --device-name TEXT         Name of the ring device  [required]
>   -t, --type [Mechanical|Digital|Not Present]  Set the in-home chime type.
>   -e, --enabled [True|False]      Enable or disable the in-home chime.
>   -d, --duration [0|1|2|3|4|5|6|7|8|9|10] Set the in-home chime duration, in seconds.
>   --help                          Show this message and exit.
>

### CODE REVIEW

File: ring_doorbell/cli.py [hunk 1]
```diff
@@ -653,6 +653,84 @@ async def _get_events(device, max_count):
...
+    help="Enable or disable the in-home chime.",
+)
+@click.option(
+    "--duration",
+    "-d",
+    default=None,
+    required=False,
+    type=click.Choice([str(i) for i in range(11)]),
+    help="Set the in-home chime duration, in seconds.",
+)
```
Comment (sdb9696) [inline ring_doorbell/cli.py:689]:
```suggestion
@click.argument("duration", type=click.IntRange(0, 10), default=None, required=False)
```

File: ring_doorbell/cli.py [hunk 2]
```diff
@@ -653,6 +653,84 @@ async def _get_events(device, max_count):
...
+    help="Set the in-home chime type.",
+)
+@click.option(
+    "--enabled",
+    "-e",
+    default=None,
+    required=False,
+    type=click.Choice(["True", "False"], case_sensitive=False),
+    help="Enable or disable the in-home chime.",
+)
```
Comment (sdb9696) [inline ring_doorbell/cli.py:681]:
```suggestion
@click.argument("enable", type=click.BOOL, default=None, required=False)
)
```

File: ring_doorbell/cli.py [hunk 3]
```diff
@@ -653,6 +653,84 @@ async def _get_events(device, max_count):
...
+    help="Name of the ring device",
+)
+@click.option(
+    "--type",
+    "-t",
+    default=None,
+    required=False,
+    type=click.Choice(["Mechanical", "Digital", "Not Present"], case_sensitive=False),
+    help="Set the in-home chime type.",
+)
```
Comment (sdb9696) [inline ring_doorbell/cli.py:673]:
```suggestion
@click.argument("duration", type=click.Choice(["Mechanical", "Digital", "Not Present"], default=None, required=False)
```

File: ring_doorbell/cli.py [hunk 4]
```diff
@@ -653,6 +653,84 @@ async def _get_events(device, max_count):
     return None
 
 
+@cli.command()
+@pass_ring
+@click.pass_context
```
Comment (sdb9696) [inline ring_doorbell/cli.py:658]:
```suggestion
@cli.group(invoke_without_command=True)
@pass_ring
@click.pass_context
async def in_home_chime(ctx, dev):
    """Show and modify in home chime settings."""
    if ctx.invoked_subcommand is None:
        return await ctx.invoke(show)
```

**Review by sdb9696 — CHANGES_REQUESTED** [review]
This is a great PR, thanks for the effort!

I would suggest that this would be better implemented as a group of related commands rather than options (also using the `click.BOOL` and `click.IntRange` types as per the comments.  This way the cli would look like:
```
ring-doorbell --device "Front Door" in-home-chime show # Shows all the settings
ring-doorbell --device "Front Door" in-home-chime # Defaults to show the settings
ring-doorbell --device "Front Door" in-home-chime enabled # Shows the enable state
ring-doorbell --device "Front Door" in-home-chime enabled y # Enables/Disables (from the docs: The string values “1”, “true”, “t”, “yes”, “y”, and “on” convert to True. “0”, “false”, “f”, “no”, “n”, and “off” convert to False.)
ring-doorbell --device "Front Door" in-home-chime duration # Shows the duration
ring-doorbell --device "Front Door" in-home-chime duration 5 # Sets the duration
ring-doorbell --device "Front Door" in-home-chime type # Shows the type
ring-doorbell --device "Front Door" in-home-chime duration Mechanical # Sets the type
```
You can set the device on the context in the main group command and `pass_dev` to the sub commands.

File: ring_doorbell/cli.py [hunk 5]
```diff
@@ -653,6 +653,84 @@ async def _get_events(device, max_count):
...
+            await device.async_set_existing_doorbell_type(2)
+            echo(f"{device_name}'s in-home chime type has been set to {type}")
+    if enabled:
+        if enabled and enabled.lower() == "true":
+            await device.async_set_existing_doorbell_type_enabled(True)
+            echo(f"{device_name}'s in-home chime has been enabled")
+        elif enabled and enabled.lower() == "false":
+            await device.async_set_existing_doorbell_type_enabled(False)
+            echo(f"{device_name}'s in-home chime has been disabled")
+    if duration:
```
Comment (sdb9696) [inline ring_doorbell/cli.py:727]:
```suggestion
@in_home_chime.group()
@pass_dev
@click.pass_context
async def duration(ctx, dev, duration):
    """Show and modify in home chime duration."""
    if duration:
```

**Comment by sdb9696** [comment]
Hey just a heads up that I’ve converted the project to use `uv` instead of `poetry`. If you merge with master you’ll need to `pipx install uv` and run things with `uv run` instead of `poetry run` but other that shouldn’t be much impact.

**Comment by briangoldstein** [comment]
@sdb9696 thank you! My concern with moving them from options to arguments is that it causes the help output to not be as helpful. Great call on moving `duration` to inter. However for `enabled`, I like keeping it locked to a True/False choice type so that it also displays in the help.  

**Help w/ options:**
Screenshot 2024-09-08 at 12 18 32 PM

**Help w/ arguments:**
Screenshot 2024-09-08 at 12 16 47 PM

**Comment by sdb9696** [comment]
> @sdb9696 thank you! My concern with moving them from options to arguments is that it causes the help output to not be as helpful. Great call on moving `duration` to inter. However for `enabled`, I like keeping it locked to a True/False choice type so that it also displays in the help.


Hey @briangoldstein, apologies for the delay in coming back to you.  I've pushed [a commit](https://github.com/python-ring-doorbell/python-ring-doorbell/pull/427/commits/3358d79c1109cf825fd6b0cc0e7e37c2f258026e) to your branch that hopefully explains what I'm trying to achieve.  Essentially I'd like to keep the use of options as light as possible and stick to arguments for the most important part of a command. The help can go in the docstring and I think it looks ok.

Some of the other commands like `motion_detection` aren't currently doing this but they will be moved over eventually.

The downside is you can't set multiple in-home-chime settings in one go but I think that's ok.

There's also some other clean up in there, let me know your thoughts.

**Comment by sdb9696** [comment]
Hi @briangoldstein, unless you object and want to revert the commit I added and discuss further, I'll merge this PR in the next day or so.

**Pull request merged**